snapcraft/_store.py

```diff
@@ -417,27 +431,90 @@ def push(snap_filename, release_channels=None):
     with _requires_login():
         store.push_precheck(snap_name)
 
-    with _requires_login():
-        tracker = store.upload(snap_name, snap_filename)
+    snap_cache = cache.SnapCache(project_name=snap_name)
+    source_snap = snap_cache.get_latest(snap_name)
+
+    sha3_384_available = hasattr(hashlib, 'sha3_384')
+
+    if os.environ.get('DELTA_UPLOADS_EXPERIMENTAL') and \
+       sha3_384_available and source_snap:
+        try:
+            result = _push_delta(snap_name, snap_filename, source_snap)
+        except (StoreDeltaApplicationError, storeapi.errors.StorePushError):
+            logger.warning('Unable to push delta, '
+                           'falling back to pushing full snap.')
```

Comment:
@kyrofa Yes, afaict catching StorePushError handles that case (anything other than the 202).  We could differentiate between a store error and a delta generation error and tell the user, but I'm not sure how useful to them that would be?